File: ViewBase/Source/Converters/BoolToHiddenVisibilityConverter.vb
﻿Imports System.Globalization
Imports System.Windows
Imports System.Windows.Data

Public Class BoolToHiddenVisibilityConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        If value Is Nothing Then
            Return Visibility.Hidden
        End If
        If value.GetType Is GetType(Boolean) Then
            Dim bool = CType(value, Boolean)
            Return If(bool, Visibility.Visible, Visibility.Hidden)
        End If
        Return Visibility.Hidden
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        Throw New NotImplementedException()
    End Function
End Class

File: BaseView/src/Converters/BoolToHiddenVisibilityConverter.cs
﻿using System;
using System.Globalization;
using System.Windows;
using System.Windows.Data;

namespace Wells.BaseView.Converters
{
    public class BoolToHiddenVisibilityConverter : IValueConverter
    {
        object IValueConverter.Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value == null) { return Visibility.Hidden; }
            if (value.GetType() == typeof(bool)) { return (bool)value ? Visibility.Visible : Visibility.Hidden; }
            return Visibility.Hidden;
        }

        object IValueConverter.ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}
